cloud-aws-common/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsSessionCredentialClientTest.java

```diff
@@ -0,0 +1,87 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import com.sequenceiq.cloudbreak.cloud.aws.common.view.AwsCredentialView;
+
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sts.StsClient;
+
+@ExtendWith(MockitoExtension.class)
+class AwsSessionCredentialClientTest {
+
+    @Mock
+    private AwsDefaultZoneProvider awsDefaultZoneProvider;
+
+    @Mock
+    private AwsEnvironmentVariableChecker awsEnvironmentVariableChecker;
+
+    @Mock
+    private AwsCredentialView awsCredentialView;
+
+    @InjectMocks
+    @Spy
+    private AwsSessionCredentialClient underTest;
+
+    @Test
+    void testAwsSecurityTokenServiceClientWhenFipsEnabledAndGovCloudCredential() {
+        String defaultRegion = setUpMocks(Boolean.TRUE, Boolean.TRUE);
+
+        StsClient actual = underTest.awsSecurityTokenServiceClient(awsCredentialView);
+
+        Assertions.assertNotNull(actual);
+        verify(underTest, times(0)).getEndpointConfiguration(defaultRegion);
+    }
+
+    @Test
+    void testAwsSecurityTokenServiceClientWhenFipsEnabledAndNotAGovCloudCredential() {
+        String defaultRegion = setUpMocks(Boolean.TRUE, Boolean.FALSE);
+
+        StsClient actual = underTest.awsSecurityTokenServiceClient(awsCredentialView);
+
+        Assertions.assertNotNull(actual);
+        verify(underTest, times(1)).getEndpointConfiguration(defaultRegion);
+    }
+
+    @Test
+    void testAwsSecurityTokenServiceClientWhenFipsIsNotEnabledAndNotAGovCloudCredential() {
+        String defaultRegion = setUpMocks(Boolean.FALSE, Boolean.FALSE);
+
+        StsClient actual = underTest.awsSecurityTokenServiceClient(awsCredentialView);
+
+        Assertions.assertNotNull(actual);
+        verify(underTest, times(1)).getEndpointConfiguration(defaultRegion);
+    }
+
+    @Test
+    void testAwsSecurityTokenServiceClientWhenFipsIsNotEnabledAndGovCloudCredential() {
+        String defaultRegion = setUpMocks(Boolean.FALSE, Boolean.TRUE);
+
+        StsClient actual = underTest.awsSecurityTokenServiceClient(awsCredentialView);
+
+        Assertions.assertNotNull(actual);
+        verify(underTest, times(1)).getEndpointConfiguration(defaultRegion);
+    }
+
+    private String setUpMocks(boolean fipsEnabled, boolean onGovCloud) {
+        String defaultRegion = Region.EU_CENTRAL_1.toString();
+        ReflectionTestUtils.setField(underTest, "fipsEnabled", fipsEnabled);
+        when(awsEnvironmentVariableChecker.isAwsAccessKeyAvailable(awsCredentialView)).thenReturn(Boolean.FALSE);
+        when(awsDefaultZoneProvider.getDefaultZone(awsCredentialView)).thenReturn(defaultRegion);
+        if (fipsEnabled) {
+            when(awsCredentialView.isGovernmentCloudEnabled()).thenReturn(onGovCloud);
+        }
```

Comment:
No, I think this stubbing should be done independently of `fipsEnabled`. This way the `fipsEnabled == false && onGovCloud == true` combination is essentially ignored and reduced to both being `false`. In other words, the current logic makes `testAwsSecurityTokenServiceClientWhenFipsIsNotEnabledAndGovCloudCredential()` behave as if it was `testAwsSecurityTokenServiceClientWhenFipsIsNotEnabledAndNotAGovCloudCredential()`.